File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        numstack = []
        
        for i in tokens:
            if i == "+" or i == "-" or i == "/" or i == "*":
                num2 = numstack.pop()
                num1 = numstack.pop()
                if i == "+":
                    numstack.append(num1 + num2)
                elif i == "-":
                    numstack.append(num1 - num2)
                elif i == "*":
                    numstack.append(num1 * num2)
                else:
                    numstack.append(int(num1 / num2))
            else:
                numstack.append(int(i))
        
        return numstack[0]
                    
            

File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        stack<int> nums;

        for (auto i : tokens) {
            if (i == "+" || i == "-" || i == "/" || i == "*") {
                int n1 = nums.top();
                nums.pop();
                int n2 = nums.top();
                nums.pop();
                
                if (i == "+") {
                    nums.push(n2 + n1);
                }

                if (i == "-") {
                    nums.push(n2 - n1);
                }

                if (i == "/") {
                    nums.push(n2 / n1);
                }

                if (i == "*") {
                    nums.push(n2 * n1);
                }
            }

            else {
                nums.push(stoi(i));
            }
        }
        return nums.top();
    }
};